integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonAlterTableCompactionCommand.scala

```diff
@@ -159,7 +161,14 @@ case class CarbonAlterTableCompactionCommand(
     // if system level compaction is enabled then only one compaction can run in the system
     // if any other request comes at this time then it will create a compaction request file.
     // so that this will be taken up by the compaction process which is executing.
-    if (!isConcurrentCompactionAllowed) {
+    if (carbonTable.isChildDataMap) {
+      carbonLoadModel.setCompactionType(alterTableModel.compactionType.toUpperCase match {
```

Comment:
change `setCompactionType` to accept enum instead of String

Reply:
done